src/opm/input/eclipse/Schedule/Well/WellKeywordHandlers.cpp

```diff
@@ -0,0 +1,916 @@
+#include <opm/input/eclipse/Schedule/Well/NameOrder.hpp>
+#include <opm/input/eclipse/Schedule/Well/WDFAC.hpp>
+#include <opm/input/eclipse/Schedule/Well/Well.hpp>
+#include <opm/input/eclipse/Schedule/Well/WellConnections.hpp>
+#include <opm/input/eclipse/Schedule/Well/WellEconProductionLimits.hpp>
+#include <opm/input/eclipse/Schedule/Well/WellTestConfig.hpp>
+#include <opm/input/eclipse/Schedule/Well/WListManager.hpp>
+#include <opm/input/eclipse/Schedule/Well/WVFPDP.hpp>
+#include <opm/input/eclipse/Schedule/Well/WVFPEXP.hpp>
+
+#include <opm/input/eclipse/Deck/DeckKeyword.hpp>
+
+#include "../HandlerContext.hpp"
+
+#include <fmt/format.h>
+
+namespace Opm {
+
+namespace {
+
+/*
+  The function trim_wgname() is used to trim the leading and trailing spaces
+  away from the group and well arguments given in the WELSPECS
+  keyword. If the deck argument contains a leading or trailing space that is
+  treated as an input error, and the action taken is regulated by the setting
+  ParseContext::PARSE_WGNAME_SPACE.
+
+  Observe that the spaces are trimmed *unconditionally* - i.e. if the
+  ParseContext::PARSE_WGNAME_SPACE setting is set to InputError::IGNORE that
+  means that we do not inform the user about "our fix", but it is *not* possible
+  to configure the parser to leave the spaces intact.
+*/
+std::string trim_wgname(const DeckKeyword& keyword,
+                        const std::string& wgname_arg,
+                        const ParseContext& parseContext,
+                        ErrorGuard& errors)
+{
+    std::string wgname = trim_copy(wgname_arg);
+    if (wgname != wgname_arg)  {
+        const auto& location = keyword.location();
+        std::string msg_fmt = fmt::format("Problem with keyword {{keyword}}\n"
+                                          "In {{file}} line {{line}}\n"
+                                          "Illegal space in {} when defining WELL/GROUP.", wgname_arg);
+        parseContext.handleError(ParseContext::PARSE_WGNAME_SPACE, msg_fmt, location, errors);
+    }
+    return wgname;
+}
+
+void handleWCONHIST(HandlerContext& handlerContext)
+{
+    for (const auto& record : handlerContext.keyword) {
+        const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
+        const auto well_names = handlerContext.wellNames(wellNamePattern, false);
+
+        const Well::Status status = WellStatusFromString(record.getItem("STATUS").getTrimmedString(0));
+
+        for (const auto& well_name : well_names) {
+            handlerContext.updateWellStatus(well_name, status,
+                                            handlerContext.keyword.location());
+
+            std::optional<VFPProdTable::ALQ_TYPE> alq_type;
+            auto well2 = handlerContext.state().wells.get( well_name );
+            const bool switching_from_injector = !well2.isProducer();
+            auto properties = std::make_shared<Well::WellProductionProperties>(well2.getProductionProperties());
```

Comment:
You should **probably** `#include <memory>` for `std::make_shared<>()` here.

Reply:
indeed